torch/_dynamo/testing.py

```diff
@@ -317,23 +317,24 @@ def _fn(*args, **kwargs):
 
 
 def make_test_cls_with_patches(cls, cls_prefix, fn_suffix, *patches, xfail_prop=None):
-    class DummyTestClass(cls):
-        pass
-
-    DummyTestClass.__name__ = f"{cls_prefix}{cls.__name__}"
+    DummyTestClass = type(f"{cls_prefix}{cls.__name__}", cls.__bases__, {})
     DummyTestClass.__qualname__ = DummyTestClass.__name__
 
-    for name in dir(cls):
+    for name in cls.__dict__.keys():
         if name.startswith("test_"):
             fn = getattr(cls, name)
             if not callable(fn):
+                setattr(DummyTestClass, name, getattr(cls, name))
                 continue
             new_name = f"{name}{fn_suffix}"
             new_fn = _make_fn_with_patches(fn, *patches)
             new_fn.__name__ = new_name
             if xfail_prop is not None and hasattr(fn, xfail_prop):
                 new_fn = unittest.expectedFailure(new_fn)
             setattr(DummyTestClass, new_name, new_fn)
+        # NB: Doesn't handle slots correctly, but whatever
+        elif name not in DummyTestClass.__dict__:
```

Comment:
```suggestion
        elif hasattr(DummyTestClass, name):
```

Would that solve the slot problem?

Reply:
Is there a preference here for has vs not has? I assumed we would want to copy over the things that don't exist